documentation/docs/05-misc/02-testing.md

```diff
@@ -2,7 +2,216 @@
 title: Testing
 ---
 
-- component testing basics
-- rune testing basics
-- vitest setup
-- e2e
+Testing helps you write and maintain your code and guard against regressions. Testing frameworks help you with that, allowing you to describe assertions or expectations about how your code should behave. Svelte is unopinionated about which testing framework you use — you can write unit tests, integration tests, and end-to-end tests using solutions like Vitest, Jasmine, Cypress and Playwright.
+
+## Unit and integration testing using Vitest
+
+Unit tests allow you to test small isolated parts of your code. Integration tests allow you to test parts of your application to see if they work together. If you're using Vite (including via SvelteKit), we recommend using [Vitest](https://vitest.dev/).
+
+To get started, install Vitest:
+
+```bash
+npm install -D vitest
+```
+
+Then adjust your `vite.config.js`:
+
+```diff
+/// file: vite.config.js
+- import { defineConfig } from 'vite';
++ import { defineConfig } from 'vitest/config';
+
+export default defineConfig({ /* ... */ })
+```
+
+You can now write unit tests for code inside your `.js/.ts` files:
+
+```js
+/// file: counter.svelte.test.js
+import { flushSync } from 'svelte';
+import { expect, test } from 'vitest';
+import { multiplier } from './multiplier.js';
+
+test('Multiplier', () => {
+	let double = multiplier(0, 2);
+
+	expect(double.value).toEqual(0);
+
+	double.set(5);
+
+	expect(double.value).toEqual(10);
+});
+```
+
+### Using runes inside your test files
+
+It is possible to use runes inside your test files. First ensure your bundler knows to route the file through the Svelte compiler before running the test by adding `.svelte` to the filename (e.g `multiplier.svelte.test.js`). After that, you can use runes inside your tests.
+
+```js
+/// file: counter.svelte.test.js
+import { flushSync } from 'svelte';
+import { expect, test } from 'vitest';
+import { multiplier } from './multiplier.svelte.js';
+
+test('Multiplier', () => {
+	let count = $state(0);
+	let double = multiplier(() => count, 2);
+
+	expect(double.value).toEqual(0);
+
+	count = 5;
+
+	expect(double.value).toEqual(10);
+});
+```
+
+If the tested code uses effects, you need to wrap the test inside `$effect.root` to create a scope in which these are properly captured.
```

Comment:
'in which these are properly captured' makes it sound like things might just behave a bit differently, rather than 'it straight up won't work'. probably simpler to just get rid

```suggestion
If the code being tested uses effects, you need to wrap the test inside `$effect.root`:
```